src/test/kotlin/sirius/kernel/commons/StringsTest.kt

```diff
@@ -332,4 +332,90 @@ class StringsTest {
+        assertEquals("ABCDEFG-…[truncated]…This is the important part", Strings.truncateMiddle("ABCDEFG-ABCDEFG-This is the important part", 8, 26))
+        assertEquals("ABCDEFG-".length, 8)
+        assertEquals("This is the important part".length, 26)
+        // Szenario 12: input present and length is smaller than input.length() and charactersToPreserveAtTheEnd is greater than length and length + charactersToPreserveAtTheEnd is greater than input.length()
+        // Note: in the following case it is assumed that the whole string should be preserved because charactersToPreserveAtTheEnd is greater than input.length()
+        assertEquals("ABCDEFG-ABCDEFG-This is the important part", Strings.truncateMiddle("ABCDEFG-ABCDEFG-This is the important part", 0, 260))
+        assertEquals("ABCDEFG-ABCDEFG-This is the important part", Strings.truncateMiddle("ABCDEFG-ABCDEFG-This is the important part", -10, 260))
+        // Szenario 13: input present and length is smaller than input.length() and charactersToPreserveAtTheEnd is equal to length and length + charactersToPreserveAtTheEnd is equal to input.length()
+        // Note: in the following case it is assumed that the whole string should be preserved because length + charactersToPreserveAtTheEnd is equal to input.length()
+        assertEquals("ABCDEFG-ABCDEFG-This is the important part", Strings.truncateMiddle("ABCDEFG-ABCDEFG-This is the important part", 21, 21))
+        assertEquals("ABCDEFG-ABCDEFG-This is the important part".length, 42)
+        // Szenario 14: input present and length is equal to input.length() and charactersToPreserveAtTheEnd is greater than length
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 7, 8))
+        // Szenario 15: input present and length is equal to input.length() and charactersToPreserveAtTheEnd is greater than input.length()
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 7, 8000))
+        // Szenario 16: input present and length is equal to input.length() and charactersToPreserveAtTheEnd is equal to input.length()
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 7, 7))
+        // Szenario 17: input present and length is 0 and charactersToPreserveAtTheEnd is greater than length
+        assertEquals("…[truncated]…FG", Strings.truncateMiddle("ABCDEFG", 0, 2))
+        // Szenario 18: input present and length is 0 and charactersToPreserveAtTheEnd is greater than input.length()
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 0, 10))
+        // Szenario 19: input present and length is 0 and charactersToPreserveAtTheEnd is equal to input.length()
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 0, 7))
+        // Szenario 20: input present and length is negative and charactersToPreserveAtTheEnd is greater than length
+        assertEquals("…[truncated]…DEFG", Strings.truncateMiddle("ABCDEFG", -10, 4))
```

Comment:
same as above, the replacement should not be longer than the input, just keep the input in those cases